app/models/concerns/s3_identifier.rb

```diff
@@ -0,0 +1,5 @@
+module S3Identifier
+  def s3_collection_uid
+    model.is_a?(Collection) ? model.uid : model.collection.uid
+  end
```

Comment:
A Collection or a Transcript instance can upload image, and further along transcript files.